From: Michael Froman <mfroman@mozilla.com>
Date: Mon, 19 Jan 2026 15:14:56 -0600
Subject: (tmp-cherry-pick) Revert "Fix Loss Based BWE V2 blocked by missing RR
 RTCP feedback" (baa2059a2a)

This reverts commit 00b7d8fdcd8e5f4712176ec6be797d358a698926.

Reason for revert: This seems to have broken a test:

test/scenario/video_stream_unittest.cc:267
Expected: (num_vga_frames_) > (0u), actual: 0 vs 0
Stack trace:
0x7fa730f7b92e: webrtc::test::VideoStreamTest_ResolutionAdaptsToAvailableBandwidth_Test::TestBody() @ ??:??
0x7fa6777bfb26: testing::Test::Run() @ ??:??
0x7fa6777c0b6b: testing::TestInfo::Run() @ ??:??
... Google Test internal frames ...

Original change's description:
> Fix Loss Based BWE V2 blocked by missing RR RTCP feedback
>
> Loss Based BWE V2 was being blocked by an early return when RR RTCP
> feedback was missing, even though V2 doesn't depend on RR RTCP packets.
>
> This change reorders the checks in UpdateEstimate() to:
> - Check Loss Based BWE V2 readiness first
> - Only check RR RTCP feedback for traditional loss-based BWE
>
> This allows V2 to work independently of RR RTCP while preserving
> existing behavior for traditional BWE.
>
> Bug: webrtc:463715720
> Change-Id: Ia9ed189298c73b200f8bb41a3a9c105f4ca8095e
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/428481
> Reviewed-by: Diep Bui <diepbp@webrtc.org>
> Reviewed-by: Per Kjellander <perkj@webrtc.org>
> Commit-Queue: Per Kjellander <perkj@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46337}

Bug: webrtc:463715720
Change-Id: I8471ea3db1a205276feb8fd80caf6a7fc530b562
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/429460
Commit-Queue: Tomas Gunnarsson <tommi@webrtc.org>
Reviewed-by: Diep Bui <diepbp@webrtc.org>
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/heads/main@{#46347}
---
 .../goog_cc/send_side_bandwidth_estimation.cc       | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/modules/congestion_controller/goog_cc/send_side_bandwidth_estimation.cc b/modules/congestion_controller/goog_cc/send_side_bandwidth_estimation.cc
index fbf8d7c3f5..244b3292f8 100644
--- a/modules/congestion_controller/goog_cc/send_side_bandwidth_estimation.cc
+++ b/modules/congestion_controller/goog_cc/send_side_bandwidth_estimation.cc
@@ -446,6 +446,12 @@ void SendSideBandwidthEstimation::UpdateEstimate(Timestamp at_time) {
     }
   }
   UpdateMinHistory(at_time);
+  if (last_loss_packet_report_.IsInfinite()) {
+    // No feedback received.
+    // TODO(srte): This is likely redundant in most cases.
+    ApplyTargetLimits(at_time);
+    return;
+  }
 
   if (LossBasedBandwidthEstimatorV2ReadyForUse()) {
     LossBasedBweV2::Result result =
@@ -455,13 +461,6 @@ void SendSideBandwidthEstimation::UpdateEstimate(Timestamp at_time) {
     return;
   }
 
-  if (last_loss_packet_report_.IsInfinite()) {
-    // No feedback received.
-    // TODO(srte): This is likely redundant in most cases.
-    ApplyTargetLimits(at_time);
-    return;
-  }
-
   TimeDelta time_since_loss_packet_report = at_time - last_loss_packet_report_;
   if (time_since_loss_packet_report < 1.2 * kMaxRtcpFeedbackInterval) {
     // We only care about loss above a given bitrate threshold.
